ch02-mr-intro/src/main/java/MaxTemperatureMapper.java

```diff
@@ -25,7 +26,7 @@ public void map(LongWritable key, Text value, Context context)
       airTemperature = Integer.parseInt(line.substring(87, 92));
     }
     String quality = line.substring(92, 93);
```

Comment:
Moving  `line.substring(92, 93)` into `qualityPattern.matcher` allows to avoid unnecessary String.substring call if airTemperature is missing

Reply:
I guess the JIT compiler could optimize it. But I applied your suggestion for clarification.